### PR TITLE
feat(jane): add PYTLocker for permanent YT locking with yield distribution

### PR DESCRIPTION
**Pull request by fp-crypto**
## Summary
- Add PYTLocker contract for permanently locking Pendle Yield Tokens while distributing yield
- Implement reward-per-share accounting to prevent dilution when new users deposit
- Support multiple YT markets with configurable SY and asset tokens

## Test plan
- [x] Unit tests for admin functions (addMarket, access control)
- [x] Unit tests for deposit/claim flows
- [x] Critical no-dilution tests verifying new depositors don't receive past yield
- [x] Proportional distribution tests
- [x] Multiple market tests
- [x] Fuzz tests for edge cases (22 tests passing)

### CODE REVIEW

File: src/jane/PYTLocker.sol [hunk 1]
```diff
@@ -51,267 +52,233 @@ contract PYTLocker is Ownable, ReentrancyGuard {
...
-    function setMarketMaxSupply(address yt, uint256 maxSupply) external onlyOwner {
-        if (!markets[yt].enabled) revert UnsupportedYT();
-        uint256 oldMaxSupply = marketMaxSupply[yt];
-        marketMaxSupply[yt] = maxSupply;
-        emit MarketMaxSupplyUpdated(yt, oldMaxSupply, maxSupply);
+    /// @param newMaxSupply The new cap (use type(uint256).max for unlimited)
+    function setMaxSupply(uint256 newMaxSupply) external onlyOwner {
+        uint256 oldMaxSupply = maxSupply;
+        maxSupply = newMaxSupply;
+        emit MaxSupplyUpdated(oldMaxSupply, newMaxSupply);
```
Comment (pandadefi) [inline src/jane/PYTLocker.sol]: You could write it:
```solidity
emit MaxSupplyUpdated(maxSupply, newMaxSupply);
maxSupply = newMaxSupply;
```
